msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java

```diff
@@ -2706,9 +2805,29 @@ private PrivateMessage getPvtMsgForward(PrivateMessage currentMessage, boolean i
     }
     
     private void processPvtMsgForwardSendHelper(PrivateMessage rrepMsg){
-    	Map<User, Boolean> recipients = getRecipients();
-    	
-    	prtMsgManager.sendPrivateMessage(rrepMsg, recipients, isSendEmail());
+	    if (booleanSchedulerSend) {
+		    setOpenDate(schedulerSendDateString);
+		    if (booleanSchedulerSend && (openDate == null || !openDate.after(Date.from(Instant.now())))) {
+			    setErrorMessage(getResourceBundleString(DATE_PGR_MSG_ERROR));
+			    return;
+		    }
+	    }
+	    rrepMsg.setScheduler(booleanSchedulerSend);
+
+	    Map<User, Boolean> recipients = getRecipients();
+
+	    if (booleanSchedulerSend && !rrepMsg.getDraft()) {
+		    rrepMsg.setScheduledDate(openDate);
+		    schedulerMessage(rrepMsg, isSendEmail());
+		    resetComposeContents();
```

Comment:
I was wondering if there is a missing return here, as the scheduled invocation will also create an event.